parlai/messenger/tasks/chatbot/run.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+from parlai.core.params import ParlaiParser
+from parlai.messenger.tasks.chatbot.worlds import \
+    MessengerBotChatTaskWorld
+from parlai.messenger.core.messenger_manager import MessengerManager
+from parlai.messenger.core.worlds import SimpleMessengerOverworld as \
+    MessengerOverworld
+from parlai.core.agents import create_agent, create_agent_from_shared
+import os
+import importlib
+
+
+def main():
+    argparser = ParlaiParser(True, True)
+    argparser.add_messenger_args()
+    opt = argparser.parse_args()
+    print(opt)
+    if opt['model'] == None and opt['model_file'] == None:
+        print("Model must be specified")
+        return
+    bot = create_agent(opt)
+    shared_bot_params = bot.share()
+
+    messenger_manager = MessengerManager(opt=opt)
+    messenger_manager.setup_server()
+    messenger_manager.init_new_state()
+
+    def get_overworld(agent):
+        return MessengerOverworld(None, agent)
+
+    def assign_agent_role(agent):
+        agent[0].disp_id = 'Agent'
+
+    def run_conversation(manager, opt, agents, task_id):
+        agent = agents[0]
+        this_bot = create_agent_from_shared(shared_bot_params)
```

Comment:
I'm guessing shared is successfully preventing needing to load multiple models?

Reply:
yep